# Deber 2

## Nombre: María Belén Rosero 

### Introducción

La importación de datos en Python mediante las bibliotecas Pandas y NumPy es una tarea esencial para el análisis de datos y la manipulación de información. Una forma común de acceder a datos es a través de un enlace o URL que apunta a un archivo de datos en formato CSV, Excel u otro formato utilizando Pandas y NumPy.

Pandas es una poderosa biblioteca de Python que proporciona estructuras de datos flexibles y herramientas de análisis de datos. NumPy, por otro lado, es una biblioteca fundamental para realizar operaciones numéricas eficientes en Python. Juntos, estos dos paquetes hacen que sea fácil importar, manipular y analizar datos.

### Importando datos

Para comenzar a importar datos desde un enlace URL, primero debemos asegurarnos de que las bibliotecas Pandas y NumPy estén instaladas en el entorno de Python.

In [1]:
import pandas as pd
import numpy as np

In [3]:
# Importar datos en un DataFrame de Pandas
data = pd.read_csv('https://raw.githubusercontent.com/cienciadedatos/datos-de-miercoles/master/datos/2019/2019-08-07/felicidad.csv')
data

pais  anio  escalera_vida   log_pib  soporte_social  \
0     Afghanistán  2008       3.723590  7.168690        0.450662   
1     Afghanistán  2009       4.401778  7.333790        0.552308   
2     Afghanistán  2010       4.758381  7.386629        0.539075   
3     Afghanistán  2011       3.831719  7.415019        0.521104   
4     Afghanistán  2012       3.782938  7.517126        0.520637   
...           ...   ...            ...       ...             ...   
1699     Zimbabue  2014       4.184451  7.562753        0.765839   
1700     Zimbabue  2015       3.703191  7.556052        0.735800   
1701     Zimbabue  2016       3.735400  7.538829        0.768425   
1702     Zimbabue  2017       3.638300  7.549491        0.754147   
1703     Zimbabue  2018       3.616480  7.553395        0.775388   

      expectativa_vida  libertad  generosidad  percepcion_corrupcion  \
0            50.799999  0.718114     0.177889               0.881686   
1            51.200001  0.678896     0.200178               0.850035   
2            51.599998  0.600127     0.134353               0.706766   
3            51.919998  0.495901     0.172137               0.731109   
4            52.240002  0.530935     0.244273               0.775620   
...                ...       ...          ...                    ...   
1699         52.380001  0.642034    -0.048634               0.820217   
1700         53.799999  0.667193    -0.097354               0.810457   
1701         54.400002  0.732971    -0.068105               0.723612   
1702         55.000000  0.752826    -0.069670               0.751208   
1703         55.599998  0.762675    -0.038384               0.844209   

      afecto_positivo  afecto_negativo  confianza  calidad_democracia  \
0            0.517637         0.258195   0.612072           -1.929690   
1            0.583926         0.237092   0.611545           -2.044093   
2            0.618265         0.275324   0.299357           -1.991810   
3            0.611387         0.267175   0.307386           -1.919018   
4            0.710385         0.267919   0.435440           -1.842996   
...               ...              ...        ...                 ...   
1699         0.725214         0.239111   0.566209           -0.985267   
1700         0.715079         0.178861   0.590012           -0.893078   
1701         0.737636         0.208555   0.699344           -0.900649   
1702         0.806428         0.224051   0.682647           -0.988153   
1703         0.710119         0.211726   0.550508                 NaN   

      calidad_entrega  de_escalera_pais_anio  gini_banco_mundial  \
0           -1.655084               1.774662                 NaN   
1           -1.635025               1.722688                 NaN   
2           -1.617176               1.878622                 NaN   
3           -1.616221               1.785360                 NaN   
4           -1.404078               1.798283                 NaN   
...               ...                    ...                 ...   
1699        -1.484067               2.079248                 NaN   
1700        -1.357514               2.198865                 NaN   
1701        -1.374650               2.776363                 NaN   
1702        -1.350867               2.656848                 NaN   
1703              NaN               2.498696                 NaN   

      gini_banco_mundial_promedio  
0                             NaN  
1                             NaN  
2                             NaN  
3                             NaN  
4                             NaN  
...                           ...  
1699                        0.432  
1700                        0.432  
1701                        0.432  
1702                        0.432  
1703                        0.432  

[1704 rows x 17 columns]

In [4]:
print(data.columns)

Index(['pais', 'anio', 'escalera_vida', 'log_pib', 'soporte_social',
       'expectativa_vida', 'libertad', 'generosidad', 'percepcion_corrupcion',
       'afecto_positivo', 'afecto_negativo', 'confianza', 'calidad_democracia',
       'calidad_entrega', 'de_escalera_pais_anio', 'gini_banco_mundial',
       'gini_banco_mundial_promedio'],
      dtype='object')


### Medidas de tendencia central y dispersión de la felicidad (data)

En esta sección se calculará una de las medidas de tendencia central y dispersión de la felicidad a nivel mundial para el primer y último año incluido en el set de datos.

In [7]:
PrimerAño = data['anio'].min()
PrimerAño

2005

In [26]:
data_2005 = data[data['anio'] == PrimerAño]
data_2005


pais  anio  escalera_vida    log_pib  soporte_social  \
59          Australia  2005       7.340688  10.569470        0.967892   
130           Bélgica  2005       7.262290  10.591697        0.934875   
192            Brasil  2005       6.636771   9.417240        0.882923   
258            Canadá  2005       7.418048  10.608347        0.961552   
386   República Checa  2005       6.439257  10.157401        0.918759   
397         Dinamarca  2005       8.018934  10.704770        0.972372   
440            Egipto  2005       5.167754   8.987387        0.847842   
496           Francia  2005       7.093393  10.505228        0.940338   
532          Alemania  2005       6.619550  10.537519        0.963490   
557            Greece  2005       6.006310  10.294153        0.836539   
624           Hungría  2005       5.193933  10.019172        0.929628   
667              Iran  2005       5.308190   9.656068        0.765978   
713            Italia  2005       6.853784  10.534875        0.928001   
738             Japón  2005       6.515817  10.481733        0.927712   
751          Jordania  2005       6.294660   9.082460        0.920013   
844            Líbano  2005       5.491245   9.414857        0.796278   
978            México  2005       6.580658   9.667230        0.902808   
1065     Países Bajos  2005       7.463979  10.680684        0.947358   
1141         Pakistan  2005       5.224658   8.297400        0.590946   
1218          Polonia  2005       5.587209   9.752288        0.921528   
1245          Rumania  2005       5.048648   9.587583        0.837685   
1281   Arabia Saudita  2005       7.079644  10.726314        0.867819   
1399           España  2005       7.152786  10.401288        0.961043   
1432           Suecia  2005       7.376316  10.626856        0.951470   
1532          Turquía  2005       4.718734   9.699514        0.819936   
1591      Reino Unido  2005       6.983557  10.522983        0.978840   
1642        Venezuela  2005       7.169621   9.612341        0.955278   

      expectativa_vida  libertad  generosidad  percepcion_corrupcion  \
59           71.400002  0.934973          NaN               0.390416   
130          69.900002  0.923843          NaN               0.597554   
192          63.299999  0.882186          NaN               0.744994   
258          71.300003  0.957306     0.244575               0.502681   
386          67.000000  0.865235          NaN               0.900733   
397          69.599998  0.971135          NaN               0.236522   
440          59.700001  0.817362          NaN                    NaN   
496          71.300003  0.894819          NaN               0.687851   
532          70.199997  0.846624          NaN               0.781007   
557          70.500000  0.734172          NaN               0.860563   
624          64.599998  0.696874          NaN               0.902811   
667          62.000000  0.651168          NaN               0.636490   
713          71.900002  0.802195          NaN               0.943912   
738          73.199997  0.867779          NaN               0.698930   
751          63.500000       NaN          NaN               0.669727   
844          64.599998  0.703206          NaN               0.945177   
978          66.199997  0.813745          NaN               0.764249   
1065         70.400002  0.901008          NaN               0.571342   
1141         54.200001  0.629996          NaN               0.844436   
1218         66.300003  0.782473          NaN               0.982931   
1245         64.000000  0.800121          NaN               0.956885   
1281         63.500000       NaN          NaN               0.505149   
1399         71.500000  0.916165          NaN               0.777272   
1432         71.199997  0.964395          NaN                    NaN   
1532         62.599998  0.623115          NaN               0.876999   
1591         69.900002  0.922355          NaN               0.398457   
1642         65.400002  0.83

Vamos a tomar la variable 'escalera_vida' en este caso es la variable felicidad. La medida que se interpretará es la media ya que así podemos conocer en promedio que tan feliz esta un país en la escala del 0 al 10. Mientras más alto el puntaje, mayor felicidad.

In [10]:
media_2005= data_2005['escalera_vida'].mean()
media_2005 

6.446164272449635

Una media de felicidad de 6.446 en una escala de 0 a 10 sugiere que, en promedio, las personas en el grupo o población que se está evaluando tienen un nivel de felicidad ligeramente por encima del punto medio de la escala, podríamos decir que es una media moderada de felicidad, a continuación compararemos con el último año para saber si la media de la felicidad aumenta o disminuye 

In [13]:
UltimoAño = data['anio'].max()
UltimoAño

2018

In [11]:
data_2018 = data[data['anio'] == UltimoAño]
data_2018


pais  anio  escalera_vida   log_pib  soporte_social  \
10    Afghanistán  2018       2.694303  7.494588        0.507516   
21        Albania  2018       5.004403  9.412399        0.683592   
28        Algeria  2018       5.043086  9.557952        0.798651   
45      Argentina  2018       5.792797  9.809972        0.899912   
58        Armenia  2018       5.062449  9.119424        0.814449   
...           ...   ...            ...       ...             ...   
1654    Venezuela  2018       5.005663  9.270281        0.886882   
1667      Vietnam  2018       5.295547  8.783416        0.831945   
1678        Yemen  2018       3.057514       NaN        0.789422   
1690       Zambia  2018       4.041488  8.223958        0.717720   
1703     Zimbabue  2018       3.616480  7.553395        0.775388   

      expectativa_vida  libertad  generosidad  percepcion_corrupcion  \
10           52.599998  0.373536    -0.084888               0.927606   
21           68.699997  0.824212     0.005385               0.899129   
28           65.900002  0.583381    -0.172413               0.758704   
45           68.800003  0.845895    -0.206937               0.855255   
58           66.900002  0.807644    -0.149109               0.676826   
...                ...       ...          ...                    ...   
1654         66.500000  0.610855    -0.176156               0.827560   
1667         67.900002  0.909260    -0.039124               0.808423   
1678         56.700001  0.552726          NaN               0.792587   
1690         55.299999  0.790626     0.036644               0.810731   
1703         55.599998  0.762675    -0.038384               0.844209   

      afecto_positivo  afecto_negativo  confianza  calidad_democracia  \
10           0.424125         0.404904   0.364666                 NaN   
21           0.713300         0.318997   0.435338                 NaN   
28           0.591043         0.292946        NaN                 NaN   
45           0.820310         0.320502   0.261352                 NaN   
58           0.581488         0.454840   0.670828                 NaN   
...               ...              ...        ...                 ...   
1654         0.759221         0.373658   0.260700                 NaN   
1667         0.692222         0.191061        NaN                 NaN   
1678         0.461114         0.314870   0.308151                 NaN   
1690         0.702698         0.350963   0.606715                 NaN   
1703         0.710119         0.211726   0.550508                 NaN   

      calidad_entrega  de_escalera_pais_anio  gini_banco_mundial  \
10                NaN               1.408344                 NaN   
21                NaN               2.640531                 NaN   
28                NaN               1.973943                 NaN   
45                NaN               2.472559                 NaN   
58                NaN               2.102111                 NaN   
...               ...                    ...                 ...   
1654              NaN               2.891930                 NaN   
1667              NaN               1.676448                 NaN   
1678              NaN               2.402008                 NaN   
1690              NaN               2.783419                 NaN   
1703              NaN               2.498696                 NaN   

      gini_banco_mundial_promedio  
10                            NaN  
21                       0.303250  
28                       0.276000  
45                       0.460938  
58                       0.319250  
...                           ...  
1654                     0.497167  
1667                     0.362750  
1678                     0.357000  
1690                     0.527400  
1703                     0.432000  

[136 rows x 17 columns]

In [12]:
media_2018= data_2018['escalera_vida'].mean()
media_2018 

5.502134340650895

En comparación con el primer año, podemos ver que en promedio la felicidad disminuyó y eso puede deberser a diferentes aspectos emocionales, sociales, económicos, políticos, etc. Que afectan de una a otra manera a la población.


#### Medida de dispersión

De la misma manera calcularemos la medida de dispersión, en este caso la desviación estándar del primer y último año

In [14]:
ds_2005= data_2005['escalera_vida'].std()
ds_2005 

0.9191426322726483

Una desviación estándar de 0.919 en una escala de 0 a 10 sugiere que los valores de felicidad tienden a estar relativamente cerca de la media, lo que indica una baja dispersión o variabilidad en los datos. Una baja desviación estándar sugiere que las puntuaciones de felicidad están agrupadas más cerca de la media. En otras palabras, la mayoría de las personas en el grupo o población tienen niveles de felicidad que son bastante similares entre sí, lo que indica una mayor consistencia en los niveles de felicidad dentro de la población. Ahora, compararemos con la desviación estándar en el último año.

In [15]:
ds_2018= data_2018['escalera_vida'].std()
ds_2018 

1.1034612436939355

En el último año podemos ver que, la desviación estándar aumentó y esto indica que los niveles de felicidad se han vuelto más diversos y que hay una mayor variabilidad en las puntuaciones de felicidad en comparación con el primer año.

Este aumento en la desviación estándar podría deberse a una serie de factores, como cambios en las circunstancias económicas, sociales o personales que hayan afectado a la población en ese período de tiempo. En términos generales, una desviación estándar más alta indica una mayor dispersión de datos alrededor de la media, lo que sugiere que los niveles de felicidad en la población se han vuelto más heterogéneos en el último año.

### Comparación de países

Comparación de valores a nivel mundial con dos países de su elección en el primer año reportado y el último año reportado.

In [81]:
paises_2005 = data_2005['pais'].unique()
paises_2005

array(['Australia', 'Bélgica', 'Brasil', 'Canadá', 'República Checa',
       'Dinamarca', 'Egipto', 'Francia', 'Alemania', 'Greece', 'Hungría',
       'Iran', 'Italia', 'Japón', 'Jordania', 'Líbano', 'México',
       'Países Bajos', 'Pakistan', 'Polonia', 'Rumania', 'Arabia Saudita',
       'España', 'Suecia', 'Turquía', 'Reino Unido', 'Venezuela'],
      dtype=object)

In [82]:
paises_2018 = data_2018['pais'].unique()
paises_2018

array(['Afghanistán', 'Albania', 'Algeria', 'Argentina', 'Armenia',
       'Australia', 'Austria', 'Azerbaijan', 'Bangladesh', 'Bielorrusia',
       'Bélgica', 'Benin', 'Bolivia', 'Bosnia y Herzegovina', 'Botswana',
       'Brasil', 'Bulgaria', 'Burkina Faso', 'Burundi', 'Cambodia',
       'Camerún', 'Canadá', 'Chad', 'Chile', 'China', 'Colombia',
       'Comoros', 'Congo (Brazzaville)', 'Costa Rica', 'Croacia',
       'Chipre', 'República Checa', 'Dinamarca', 'República Dominicana',
       'Ecuador', 'Egipto', 'El Salvador', 'Estonia', 'Etiopía',
       'Finlandia', 'Francia', 'Gabón', 'Gambia', 'Georgia', 'Alemania',
       'Greece', 'Guatemala', 'Guinea', 'Haití', 'Honduras', 'India',
       'Indonesia', 'Iran', 'Irlanda', 'Israel', 'Italia',
       'Costa de Marfil', 'Japón', 'Jordania', 'Kazakhstan', 'Kenia',
       'Kosovo', 'Kyrgyzstan', 'Laos', 'Latvia', 'Líbano', 'Liberia',
       'Libya', 'Lituania', 'Luxemburgo', 'Macedonia', 'Madagascar',
       'Malawi', 'Malasia', 'Mali',

Del array, vamos a tomar 2 países 

In [84]:
Venezuela = 'Venezuela'
Iran = 'Iran'

In [85]:
datos_Venezuela_2005 = data_2005[data_2005['pais'] == Venezuela]
datos_Iran_2005 = data_2005[data_2005['pais'] == Iran]

In [86]:
datos_Venezuela_2018 = data_2018[data_2018['pais'] == Venezuela]
datos_Iran_2018 = data_2018[data_2018['pais'] == Iran]

In [88]:
FELICIDAD_venezuela=datos_Venezuela_2005.iloc[:, 2]
FELICIDAD_venezuela

1642    7.169621
Name: escalera_vida, dtype: float64

In [89]:
FELICIDAD_iran=datos_Iran_2005.iloc[:, 2]
FELICIDAD_iran

667    5.30819
Name: escalera_vida, dtype: float64

A nivel mundial en 2005 tenemos que el promedio de la felicidad es de 6.45 y en comparación con Venezuela esta por encima de la media, la población en Venezuela en este año sentía mejores condiciones para estar felicices. Por otro lado, Iran esta por debajo de la media y una de las razones puede ser a la tensión que hubo en la crisis nuclear de Iran en 2005, lo que causó que la población experimente una inflación creciente, la devaluación de la moneda y una disminución de la calidad de vida en general. Las restricciones financieras hicieron que el acceso a bienes y servicios esenciales fuera más difícil para muchas personas.

In [90]:
FELICIDAD_venezuela_2018=datos_Venezuela_2018.iloc[:, 2]
FELICIDAD_venezuela_2018

1654    5.005663
Name: escalera_vida, dtype: float64

In [91]:
FELICIDAD_iran_2018=datos_Iran_2018.iloc[:, 2]
FELICIDAD_iran_2018

677    4.278118
Name: escalera_vida, dtype: float64

En 2018 podemos ver que el promedio de la felicidad disminuyó a 5.50, y del mismo modo los países de Venezuela e Irán. Debido a una crisis económica y política que provocó la hiperinflación, un aumento en la migración, Venezuela disminuyó a 5 que está por debajo de la felicidad en 2005. Del mismo modo, en Iran la crisis política y económica absorbe a la clase media de este país reduce en su escala de la felicidad, vale recalcar que Iran es un país en donde se rige a la religión, lo que estó también provoca en la población disminución de la felicidad.